src/python/geoclaw/xarray_backends.py

```diff
@@ -0,0 +1,588 @@
+    # default behavior assigns no coordinate system.
+
+    # An example of a fgmax grid.
+    filename = "_output/fgmax0001.txt"
+    ds = xr.open_dataset(filename, engine=FGMaxBackend, backend_kwargs={'epsg':epsg_code})
+
+
+Dimensions:
+
+Files opened with FGOutBackend will have dimensions (time, y, x).
+Files opened with FGMaxBackend will have dimensions (y, x).
+
+Variable naming:
+
+For fixed grid files, the dataset will have the elements of q specified
+by qmap and q_out_vars. This may be as many as the full list described below or
+as few as specified by the user.
+
+For fixed grid geoclaw files, the full set of variables is:
+- h
+- hu
+- hv
+- eta
+- B
+
+For fixed grid geoclaw-bouss files, the full set of variables is:
+- h
+- hu
+- hv
+- huc
+- hvc
+- eta
+- B
+
+Fixed grid dclaw files, the full set of variables is:
+- h
+- hu
+- hv
+- hm
+- pb
+- hchi
+- delta_a
+- eta
+- B
+
+Depending on the number of variables specified in the setrun.py fgmax files will
+have a portion of the following variables:
+
+If rundata.fgmax_data.num_fgmax_val == 1
+
+- arrival_time, Wave arrival time (based on eta>sea_level + fg.arrival_tol)
+- h_max, Maximum water depth
+- eta_max, Maximum water surface elevation
+- h_max_time, Time of maximum water depth
+- B, Basal topography at the first time fgmax first monitored maximum amr level
+- level, Maximum amr level
+
+If rundata.fgmax_data.num_fgmax_val == 2:
+
+- s_max, Maximum velocity
+- s_max_time, Time of maximum velocity
+
+If rundata.fgmax_data.num_fgmax_val == 5:
+
+- hs_max, Maximum momentum
+- hs_max_time, Time of maximum momentum
+- hss_max, Maximum momentum flux
+- hss_max_time, Time of maximum momentum flux
+- h_min, Minimum depth
+- h_min_time, Time of minimum depth
+
+
+See the following links for additional information about xarray Backends.
+
+- https://docs.xarray.dev/en/stable/generated/xarray.backends.BackendEntrypoint.html#xarray.backends.BackendEntrypoint
+- https://docs.xarray.dev/en/stable/generated/xarray.open_dataset.html
+"""
+
+import os
+
+import numpy as np
+
+try:
+    import rioxarray  # activate the rio accessor
+    import xarray as xr
+except ImportError:
+    raise ImportError(
+        "rioxarray and xarray are required to use the FGOutBackend and FGMaxBackend"
+    )
+
+from clawpack.geoclaw import fgmax_tools, fgout_tools
+from xarray.backends import BackendEntrypoint
+
+_qelements_dclaw = ["h", "hu", "hv", "hm", "pb", "hchi", "delta_a", "eta", "B"]
+_qelements_geoclaw = ["h", "hu", "hv", "eta", "B"]
+_qelements_geoclaw_bouss = ["h", "hu", "hv", "huc", "hvc", "eta", "B"]
+_qunits = {
+    "h": "meters",
+    "hu": "meters squared per second",
+    "hv": "meters squared per second",
+    "hm": "meters",
+    "pb": "newton per meter squared",
+    "hchi": "meters",
+    "delta_a": "meters",
+    "eta": "meters",
+    "B": "meters",
+    "huc": "meters squared per second",
+    "hvc": "meters squared per second",
+}
```

Comment:
@rjleveque I did have a question about whether the units for huc and hvc were correct. I don't know how the corrections are applied (if they are added or multiplied or something else, so I wasn't sure what to put).

Reply:
Thanks for the units information. I've changed it to 'varies'.

'bdif' is better than 'delta_a'. This has been called different things at different times. At present it is also called 'b_eroded' internally and in the dclaw data.py docstrings (I'll change that for consistency).

I've changed delta_a to bdif as well. Good catch.